src/groupByKey.ts

```diff
@@ -2,6 +2,25 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  const mainObj: GroupsMap<T> = {};
+
+  for (let i: number = 0; i < items.length; i++) {
```

Comment:
Consider using a more descriptive variable name instead of 'i'. For example, 'index' would improve code readability.